### ISSUE TITLE
usage of maintain aspect ratio

### ISSUE THREAD

**Issue by 7yl4r** [titu1994/Neural-Style-Transfer#23]
I may be doing something wrong here, but I also see the latest commit mentions a change to maintain_aspect_ratio. Using the latest build all images are coming out square. I've tried passing --maintain_aspect_ratio 1 as well, but no luck. 

If someone can confirm it isn't just me being dumb, [this line in the last commit](https://github.com/titu1994/Neural-Style-Transfer/commit/56eed72f553649648db53bff0fd88a9c83052ac3#diff-a37df1f394eb07b1c1d5fe4cfb8f54b4R587) may be the culprit.

**Comment by titu1994** [comment]
@7yl4r The last commit did change the behaviour of the maintain_aspect_ratio parameter, but I made it so that the image that is generated at the end of the script will always maintain the aspect ratio of the original content image. The parameter now acts as a flag to load a 400x640 image (if set to 1) or else a 400x400 image as the content and style image (assuming that image_size = 400)

The replacement of `if maintain_aspect_ratio and (not rescale_image)` with `if not rescale_image` basically means that if you aren't going to upscale the image to the original dimensions (say 1080p using bicubic upsampling), then at least maintain the aspect ratio (say 400x640)

Simply put, as long as you aren't providing a square image and aren't upscaling to full size of the original image, it should maintain the aspect ratio in the generated image. Can you provide me with some sample content images on which I can test this?

**Comment by 7yl4r** [comment]
Here is what I get from the first iteration of 

`python INetwork.py --image_size 200 ./images/inputs/content/tubingen.jpg ./images/inputs/style/starry_night.jpg ./test`

![test_at_iteration_1](https://cloud.githubusercontent.com/assets/1051390/20779210/fcd7f0d2-b740-11e6-8fc0-c79261046192.png)

**Comment by titu1994** [comment]
Can you see if there is a line such as "Rescaling Image to (400, 640)" after each iteration ?

**Comment by titu1994** [comment]
Try with the argument : --rescale_image "False"

**Comment by 7yl4r** [comment]
It says rescaling to 200x200

**Comment by titu1994** [comment]
Can you copy paste the Network.py and INetwork.py code from the repo into your local copies and run it again? Because this is the results I am getting : 

"D:/Users/Yue/PycharmProjects/Neural Style Transfer/INetwork.py" D:\Yue\Pictures\hoovertowernight.jpg D:\Yue\Pictures\Art\starry_night.jpg "D:\Yue\Desktop\Neural Art\test" --image_size 200

```
Model loaded.
Starting iteration 1 of 10
Current loss value: 135229216.0  Improvement : 0.000 %
Rescaling Image to (200, 380)
Image saved as D:\Yue\Desktop\Neural Art\test_at_iteration_1.png
```

![test_at_iteration_1](https://cloud.githubusercontent.com/assets/3048602/20779447/97466794-b79a-11e6-8990-3b30df079e8b.png)

**Comment by 7yl4r** [comment]
I have cloned the repo through git, and git status shows no changes from master. Adding `--rescale_image "False"` didn't have an effect.

**Comment by titu1994** [comment]
Well that is weird. The same code having different results ? What is your python version? 2.7 ? 

There shouldn't be any difference in calculating the aspect ratio between python 3 and 2 though

**Comment by 7yl4r** [comment]
Python version 2.7.12. I'm going to add some print statements and see if I can find what is going on and I'll get back to you.

**Comment by titu1994** [comment]
Yes thanks for that. Also, try adding the original line `if maintain_aspect_ratio and (not rescale_image)` to see if it makes any difference.

**Comment by 7yl4r** [comment]
I believe I've identified the issue and suggested a fix. `aspect_ratio` was being calculated using integer math, so I've suggested a cast to force floating point computation. I'm guessing the difference has to do with how command line vars are evaluated on windows vs linux.

**Comment by titu1994** [comment]
Oh ! Nice catch. If you wish to be credited for the fix, could you create a pull request ?

**Comment by titu1994** [comment]
Thanks for the fix !
